### PR TITLE
Use bindings layout for all cuda-python imports.

### LINKED ISSUE #1755: [FEA] Use new cuda-python `cuda.bindings` layout for imports
**Is your feature request related to a problem? Please describe.**
`cuda-python` recently released some API changes, renaming several components like `cuda.{cuda,cudart}` to `cuda.bindings.{cydriver,cyruntime,driver,runtime}`.

**Describe the solution you'd like**
We should adopt the new names in rmm.

**Additional context**
See also:
- https://github.com/rapidsai/build-planning/issues/117
- #1675 will need some tweaks after this change. I plan to create a new PR that includes those changes alongside the `cuda.bindings` changes.

### PR DESCRIPTION
**Pull request by bdice**
## Description
Closes #1755.

Replaces #1675.

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/rmm/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: python/rmm/rmm/_cuda/gpu.py [hunk 1]
```diff
@@ -1,14 +1,15 @@
-# Copyright (c) 2020, NVIDIA CORPORATION.
+# Copyright (c) 2020-2024, NVIDIA CORPORATION.
 
-from cuda import cuda, cudart
+import cuda.bindings.driver as driver
+import cuda.bindings.runtime as runtime
```
Comment (Matt711) [inline python/rmm/rmm/_cuda/gpu.py:4]: Maybe?
```suggestion
from cuda.bindings import driver, runtime
```

**Review by Matt711 — APPROVED** [review]

File: python/rmm/rmm/_cuda/gpu.py (hunk 1 above)
Comment (bdice) [reply python/rmm/rmm/_cuda/gpu.py:4]: Oh, good catch. I was playing with different names earlier. This is the right call!

**Review by vyasr — APPROVED** [review]
LGTM

**Pull request merged**